edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/ModuleClient.cs

```diff
@@ -2,249 +2,212 @@
 namespace Microsoft.Azure.Devices.Edge.Agent.IoTHub
 {
     using System;
-    using System.Collections.Generic;
-    using System.ComponentModel;
-    using System.Net;
-    using System.Runtime.ExceptionServices;
+    using System.Linq;
     using System.Threading;
     using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Client;
-    using Microsoft.Azure.Devices.Client.Exceptions;
-    using Microsoft.Azure.Devices.Client.Transport.Mqtt;
     using Microsoft.Azure.Devices.Edge.Agent.Core;
+    using Microsoft.Azure.Devices.Edge.Agent.IoTHub.SdkClient;
     using Microsoft.Azure.Devices.Edge.Util;
-    using Microsoft.Azure.Devices.Edge.Util.TransientFaultHandling;
+    using Microsoft.Azure.Devices.Edge.Util.Concurrency;
     using Microsoft.Azure.Devices.Shared;
     using Microsoft.Extensions.Logging;
-    using ExponentialBackoff = Microsoft.Azure.Devices.Edge.Util.TransientFaultHandling.ExponentialBackoff;
 
     public class ModuleClient : IModuleClient
     {
-        const uint DeviceClientTimeout = 30000; // ms
-        static readonly ITransientErrorDetectionStrategy TransientErrorDetectionStrategy = new ErrorDetectionStrategy();
-
-        static readonly RetryStrategy TransientRetryStrategy =
-            new ExponentialBackoff(int.MaxValue, TimeSpan.FromSeconds(2), TimeSpan.FromSeconds(60), TimeSpan.FromSeconds(4));
+        static readonly Type[] NonRetryableExceptions =
+        {
+            typeof(NullReferenceException),
+            typeof(ObjectDisposedException)
+        };
 
-        readonly Client.ModuleClient moduleClient;
+        readonly AtomicBoolean isActive = new AtomicBoolean(true);
+        readonly ISdkModuleClient inner;
+        readonly ResettableTimer inactivityTimer;
 
-        ModuleClient(Client.ModuleClient moduleClient)
+        public ModuleClient(ISdkModuleClient inner, TimeSpan idleTimeout, bool closeOnIdleTimeout)
```

Comment:
>inner [](start = 45, length = 5)

I was seeing ModuleClient as a wrapper over SDK client, why is another wrapping layer needed?